api/src/main/java/org/openmrs/api/db/hibernate/PatientSearchCriteria.java

```diff
@@ -264,6 +264,7 @@ private void addAliasForIdentifiers(Criteria criteria) {
 	 * @param identifierTypes
 	 * @param matchIdentifierExactly
 	 * @param includeVoided true/false whether or not to included voided patients
+	 * @should return a criterion when identifierTypes is not empty
```

Comment:
The test for identifier type restrictions means test if given identifier types return the expected results. You can look at the restriction here: https://github.com/hilz041/openmrs-core/blob/8ff4b3763ac8c98913e3d16c3b2b60a3e1549ebd/api/src/main/java/org/openmrs/api/db/hibernate/PatientSearchCriteria.java#L318